### ISSUE TITLE
Not working with versioned API

### ISSUE THREAD

**Issue by npflood** [superiorlu/jsonapi-swagger#2]
I have a project that is using a versioned API. Controllers and resources are structured under a "V1" namespace so that my category resource is defined like this:

```RUBY
class V1::CategoryResource < JSONAPI::Resource
	attribute :name, type: :string
	attribute :description, type: :string
end
```

When I run this: `rails g jsonapi:swagger Category`
I get **Category not support! (Jsonapi::Swagger::Error)**
Which is not unexpected because Category doesn't really exist.

However when I run this `rails g jsonapi:swagger V1::Category`
I get **undefined method `columns' for nil:NilClass (NoMethodError)**

Here's the relevant part of the trace:
```
from /usr/local/bundle/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	25: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/swagger_generator.rb:10:in `create_swagger_file'
	24: from /usr/local/bundle/gems/railties-6.0.0.rc1/lib/rails/generators/named_base.rb:24:in `template'
	23: from /usr/local/bundle/gems/railties-6.0.0.rc1/lib/rails/generators/named_base.rb:45:in `inside_template'
	22: from /usr/local/bundle/gems/railties-6.0.0.rc1/lib/rails/generators/named_base.rb:25:in `block in template'
	21: from /usr/local/bundle/gems/thor-0.20.3/lib/thor/actions/file_manipulation.rb:122:in `template'
	20: from /usr/local/bundle/gems/thor-0.20.3/lib/thor/actions/create_file.rb:25:in `create_file'
	19: from /usr/local/bundle/gems/thor-0.20.3/lib/thor/actions.rb:94:in `action'
	18: from /usr/local/bundle/gems/thor-0.20.3/lib/thor/actions/create_file.rb:60:in `invoke!'
	17: from /usr/local/bundle/gems/thor-0.20.3/lib/thor/actions/empty_directory.rb:115:in `invoke_with_conflict_check'
	16: from /usr/local/bundle/gems/thor-0.20.3/lib/thor/actions/create_file.rb:73:in `on_conflict_behavior'
	15: from /usr/local/bundle/gems/thor-0.20.3/lib/thor/actions/create_file.rb:46:in `identical?'
	14: from /usr/local/bundle/gems/thor-0.20.3/lib/thor/actions/create_file.rb:53:in `render'
	13: from /usr/local/bundle/gems/thor-0.20.3/lib/thor/actions/file_manipulation.rb:131:in `block in template'
	12: from /usr/local/lib/ruby/2.6.0/erb.rb:901:in `result'
	11: from /usr/local/lib/ruby/2.6.0/erb.rb:901:in `eval'
	10: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/templates/swagger.json.erb:266:in `template'
	 9: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/templates/swagger.json.erb:167:in `list_resource_responses'
	 8: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/templates/swagger.json.erb:77:in `properties'
	 7: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/templates/swagger.json.erb:77:in `tap'
	 6: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/templates/swagger.json.erb:78:in `block in properties'
	 5: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/templates/swagger.json.erb:78:in `each'
	 4: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/templates/swagger.json.erb:78:in `each_key'
	 3: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/templates/swagger.json.erb:79:in `block (2 levels) in properties'
	 2: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/swagger_generator.rb:121:in `columns_with_comment'
	 1: from /usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/swagger_generator.rb:121:in `tap'
/usr/local/bundle/gems/jsonapi-swagger-0.7.0/lib/generators/jsonapi/swagger/swagger_generator.rb:122:in `block in columns_with_comment': undefined method `columns' for nil:NilClass (NoMethodError)
```

**Comment by npflood** [comment]
The problem seems to be that you are inferring the table name from the argument passed to the generator instead of from the model to which the resource refers. 

Since the name of the underlying model can be defined using the `model_name` keyword  in a JSONAPI::Resource the name of the model should not be inferred from the name of the resource but rather from the resources model hints.

**Comment by zernie** [comment]
@npflood I have the same issue, and adding `model_name 'Category'` to `Api::V1::CategoryResource` doesn't help. What do? :)

**Comment by superiorlu** [comment]
i fix this namespace problem, can upgrade `0.7.1`.
@npflood  run `rails g jsonapi:swagger V1::Category`
@zernie run `rails g jsonapi:swagger Api::V1::Category`

**Comment by npflood** [comment]
Thanks! There are a few odd things that happen when you do this but nothing that can't be cleaned up. I'll start working on a some of them and see if I can't put in a few PRs.
